src/api/route-services/plots-tables.js

```diff
@@ -28,6 +27,26 @@ class PlotsTablesService {
     return tableData;
   }
 
+  async updatePlotData(experimentId, plotUuid, plotData) {
+    const marshalledData = convertToDynamoDbRecord({ ':d': plotData });
+
+    const params = {
+      TableName: this.tableName,
+      Key: {
+        experimentId: { S: experimentId }, plotUuid: { S: plotUuid },
+      },
+      UpdateExpression: 'SET plotData = :d',
+      ExpressionAttributeValues: marshalledData,
+      ReturnValues: 'UPDATED_NEW',
+    };
+
+    const dynamodb = createDynamoDbInstance();
+    const result = await dynamodb.updateItem(params).promise();
```

Comment:
Will this work reliably if an experiment does not yet have any plots and tables in DynamoDB?

I think we discussed that in that case we will do a partial plot with only the data in it, and UI will fill in the rest. Can this accomplish this?

Reply:
Yes, `updateItem` updates an existing item or create the record if it doesn't already exist.

Reply:
Yes, `updateItem` updates an existing item or create the record if it doesn't already exist.

Reply:
I'll test it out with the UI,

Reply:
@marcellp , I've updated this so that the update also includes. The DynamoDB now includes : 

```
{
  "plotType": "genesVsUmisScatterplot",
  "experimentId": "e52b39624588791a7889e39c617f669e",
  "plotData": {
    "plot2": [
      4,
      5,
      6
    ],
    "plot1": [
      11,
      2,
      3
    ]
  },
  "plotUuid": "genesVsUmisScatterplot",
  "config": {}
}
```